src/components/Navigation/components/Item/Item.tsx

```diff
@@ -230,7 +230,7 @@ export function Item({
                 key={label}
                 label={label}
                 matches={item === longestMatch}
-                onClick={onNavigationDismiss}
+                onClick={getClickHandler(item.onClick)}
```

Comment:
This is the same function all other nav item click handlers are resolved from so that the nav is dismissed on small screen if needed while also calling the item's `onClick` when provided.